misc/qlog-adapter.py

```diff
@@ -328,22 +328,32 @@ def render_ack_frame(ranges):
 def usage():
     print(r"""
 Usage:
-    python qlog-adapter.py inTrace.jsonl
+    python3 qlog-adapter.py [inTrace.jsonl]
+
+    If the argument is omitted, inTrace will be read from stdin.
 """.strip())
 
 def load_quicly_events(infile):
     events = []
-    with open(infile, "r") as fh:
-        for line in fh:
-            events.append(json.loads(line))
+    for line in infile:
+        events.append(json.loads(line))
     return events
 
 def main():
-    if len(sys.argv) != 2:
+    if len(sys.argv) > 2:
         usage()
         sys.exit(1)
 
-    (_, infile) = sys.argv
+    if len(sys.argv) == 1:
+        infile = sys.stdin
+    else:
+        (_, fn) = sys.argv
+        try:
+            infile = open(fn, "r")
+        except OSError as e:
+            print("Failed to open %s: %s" % (fn, e.strerror))
+            sys.exit(1)
```

Comment:
`sys.exit("Failed to open %s: %s" % (fn, e.strerror))` seems better to write error messages to stdout and exit with non-zero status code.

Reply:
Just learned about this syntax. Addressed in [eaae916](https://github.com/h2o/quicly/pull/508/commits/eaae916f290222d43e66ee5a303a8b2244471923)

Reply:
And it looks like `sys.exit(msg)` does seem to emit `msg` to _stderr_, which is good.